subprojects/packagefiles/libgpiod/man/meson.build

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# Copyright 2021 IBM Corp.
+
+if get_option('manpages').enabled()
+    warning('Meson lacks help2man support, unable to generate manpages')
```

Comment:
What do you mean by lacking support?

It is admittedly true that:
- you should be using custom_target, not generator
- install_man kind of sucks at the moment, you need to manually build the install_dir path without its help, e.g. https://gitlab.archlinux.org/pacman/pacman/-/blob/fc7986485cad9c93df7e59979a7e950fffdc4271/doc/meson.build#L45-47

Reply:
Hmm, I tried a few different approaches and eventually found an open issue about supporting `help2man` via `run_target()`. I stopped experimenting at that point.

I can try with `custom_target()`.

Reply:
run_target is basically the equivalent of a Makefile .PHONY target which runs arbitrary commands such as a "dist" target that makes a tarball, then scp's it to a server.

custom_target is meson's generic rule for creating a file input->output target with a manually coded compile command.

executable or library or shared_module is meson's super-smart macro rule for detecting the best compile command per platform, and using that.

Reply:
Alright, I got manpages for the tools working, but the library API manpages generated with doxygen seem a little more difficult with the upstream `Doxyfile.in`. For now I'm building them but not installing them by default.

Reply:
In my project, we somehow got it to work by instructing doxygen to emit man pages to a "man3" directory, and listing that directory name as the doxygen output.

meson manages to do the right thing if a directory name is the custom_target output, it copies the entire directory tree to the install dir.

Reply:
Yeah I saw that, but I'm not imaginative enough to make that work here without patching the upstream source. Upstream sets the output directory to 'doc' as a literal (as opposed to via `@var@`) in `Doxyfile.in`, and then we get `doc/man3`. `custom_target` barfs about not allowing path components if I do `output: [ 'doc/man3' ]`.

Reply:
You could create a doc/meson.build which tells it to output to ../ and then it creates ../doc/. I think this should work...